test/e2e/cni/cni_test.go

```diff
@@ -0,0 +1,283 @@
+					// Update aws-node env vars
+					envs := []corev1.EnvVar{
+						{Name: "WARM_IP_TARGET", Value: "0"},
+						{Name: "WARM_ENI_TARGET", Value: "2"},
+						{Name: "MAX_ENI", Value: "-1"},
+					}
+					i = 1
+					utils.UpdateDaemonSetEnvVars(ctx, f, kubeSystem, awsNodeDS, envs)
+
+					internalIP, eniLimit, ipLimit, coreDNSCount = setup(ctx, f, ns, serviceAccount, awsNodeDS, i)
+					// podCount is the remaining number of secondary IPs on the first ENI
+					podCount = int32(ipLimit - coreDNSCount)
+				})
+				It("Should have the expected aws-node prometheus metrics and be below the error rate", func() {
+					setUpPrometheus(ctx, ns, serviceAccount, testerNodeName)
+					testAWSNodePromMetrics(time.Now(), internalIP, eniLimit, ipLimit, awsNodeErrLimit)
+				})
+				It("Should have the expected testpod prometheus metrics and be below the error rate", func() {
+					setUpPrometheus(ctx, ns, serviceAccount, testerNodeName)
+
+					By("creating testpod resources")
+					testpodResources = resources.NewTestpodResources(ns.Name, serviceAccount, nodes[i].Name, testpodCount)
+					testResources = append(testResources, testpodResources)
+					testpodResources.ExpectDeploySuccessful(ctx, f, waitDuration, ns)
+
+					// Sleep for metrics to exist
+					time.Sleep(testpodSleepDuration)
+					testTestpodPromMetrics(time.Now(), testPodErrPercentLimit)
+				})
+				// With WARM_IP_TARGET=0 (default), WARM_ENI_TARGET=2, and MAX_ENI=-1(default), the Amazon VPC CNI wants
+				// two ENIs to have no used secondary IPs.
+				It("Should successfully scale ENIs with WARM_IP_TARGET=0 (default), WARM_ENI_TARGET=2 (default), and MAX_ENI=-1(default)", func() {
+					// Adding anywhere from 1 pod up to the limit for an ENI adds two ENIs up to the max ENIs allowed for the instance type.
+					nginxResources = resources.NewNginxResources(ns.Name, serviceAccount, nodes[i].Name, podCount)
+					testResources = append(testResources, nginxResources)
+					By(fmt.Sprintf("deployment (%s) with %d pods should have 2 ENIs", nginxResources.Deployment.Name, podCount))
+					nginxResources.ExpectDeploySuccessful(ctx, f, waitDuration, ns)
+					cni.TestENIInfo(ctx, f, internalIP, eniLimit, ipLimit, 3)
+
+					// And adding 1 more secondary IP adds an IP on the second ENI, resulting in a third ENI
+					By(fmt.Sprintf("scaling deployment (%s) to %d pods to get 3 ENIs", nginxResources.Deployment.Name, podCount+1))
+					nginxResources.ExpectDeploymentScaleSuccessful(ctx, f, waitDuration, ns, podCount+1)
+					cni.TestENIInfo(ctx, f, internalIP, eniLimit, ipLimit, 4)
+				})
+			})
+			AfterEach(func() {
+				for _, testResource := range testResources {
+					testResource.ExpectCleanupSuccessful(ctx, f, ns)
+				}
+			})
+		})
+	})
+})
+
+func setUpPrometheus(ctx context.Context, ns *corev1.Namespace, serviceAccount string, testerNodeName string) {
+	By("Creating prometheus resources")
+	promResources = resources.NewPromResources(ns.Name, serviceAccount, testerNodeName, 1)
```

Comment:
How do we configure prometheus for the test?  Are we defining scrape targets somewhere?  Is it possible to define a struct in aws-k8s-tester that is generic enough so that it can be passed into NewPromResources and it includes target definitions, or are they discovered dynamically?

Reply:
Prometheus is just set up at https://github.com/aws/aws-k8s-tester/blob/master/e2e/resources/prometheus.go and it scrapes anything with `"prometheus.io/scrape":"true"`.

The prow code (which should end up in a README or be another patch in here, adds `"prometheus.io/port":"61678", "prometheus.io/scrape":"true"` to the `aws-node` daemonset. Testpod has 8080 as the port: https://github.com/aws/aws-k8s-tester/blob/master/e2e/resources/testpod.go#L26